File: general-server/src/main/java/com/song/middleware/server/web/support/ApiResult.java
package com.song.middleware.server.web.support;

import java.io.Serializable;
import lombok.Getter;
import lombok.Setter;

/**
 * Created by song on 2017/8/5.
 */
@Getter
@Setter
public class ApiResult<T> implements Serializable {

    private boolean success;

    private String message;

    private T data;

    public ApiResult() {
    }

    public ApiResult(boolean success) {
        this.success = success;
    }

    public static <T> ApiResult<T> create() {
        return create(false);
    }

    public static <T> ApiResult<T> create(boolean success) {
        return new ApiResult<>(success);
    }


}

File: general-server/src/main/java/com/song/middleware/server/web/support/ApiResult.kt
package com.song.middleware.server.web.support

import java.io.Serializable

/**
 * Created by song on 2017/8/5.
 */
class ApiResult<T> : Serializable {

    var isSuccess: Boolean = false

    var message: String? = null

    var data: T? = null

    constructor() {}

    constructor(success: Boolean) {
        this.isSuccess = success
    }

    companion object {

        fun <T> create(): ApiResult<T> {
            return create(false)
        }

        fun <T> create(success: Boolean): ApiResult<T> {
            return ApiResult(success)
        }
    }
}
